docker/web-and-data/Dockerfile

```diff
@@ -8,16 +8,49 @@
 #
 # docker build -f docker/web-and-data/Dockerfile -t cbioportal-container:tag-name .
 #
-# WARNING: Be careful about publishing images generated like this publicly
-# because your .git folder is exposed in the build step. We are not sure if
-# this is a security risk: stackoverflow.com/questions/56278325
+# NOTE: the .git folder is included in the build stage, but excluded 
+# from the final image. No confidential information is exposed.
+# (see: stackoverflow.com/questions/56278325)
 FROM maven:3.5.4 as build
-COPY $PWD /cbioportal
+
+# download maven dependencies first to take advantage of docker caching
+COPY pom.xml                                     /cbioportal/
+COPY version.sh                                  /cbioportal/
 WORKDIR /cbioportal
+
+COPY business/pom.xml                         business/
+COPY core/pom.xml                             core/
+COPY db-scripts/pom.xml                       db-scripts/
+COPY model/pom.xml                            model/
+COPY persistence/pom.xml                      persistence/
+COPY persistence/persistence-api/pom.xml      persistence/persistence-api/
+COPY persistence/persistence-mybatis/pom.xml  persistence/persistence-mybatis/
+COPY scripts/pom.xml                          scripts/
+COPY security/pom.xml                         security/
+COPY security/security-spring/pom.xml         security/security-spring/
+COPY service/pom.xml                          service/
+COPY web/pom.xml                              web/
```

Comment:
It would be nice if the pom files weren't hardcoded but there might not be an easy way to do that

Reply:
The thing is that your would want them on separate lines of the Dockerfile. That would then only be possible when you create a Dockerfile via script.

Reply:
I see - thanks